nautobot/core/testing/filters.py

```diff
@@ -155,6 +170,40 @@ def test_tags_filter(self):
             qs_result = self.queryset.filter(tags=tags[0]).filter(tags=tags[1]).distinct()
             self.assertQuerysetEqualAndNotEmpty(filterset_result, qs_result)
 
+        def test_q_filter_exists(self):
+            """Test the `q` filter exists on a filterset, does not validate the filter works as expected."""
+            self.assertTrue(self.filterset.declared_filters.get("q"))
+
+        def test_q_filter_valid(self):
+            """Test the `q` filter based on attributes in `filter_predicates`."""
+            if not isinstance(self.filterset, SearchFilter):
+                self.skipTest("`q` filter is not a SearchFilter")
+            if not self.filterset.declared_filters.get("q"):
+                self.skipTest("`q` filter not implemented")
+
+            obj = self.queryset.first()
+            for attr, lookup_method in self.get_q_filter().items():
+                if lookup_method not in ["icontains", "iexact"]:
+                    # only testing icontains and iexact filter lookups
+                    continue
+                lookup = str(obj.id)
+                if not attr == "id":
```

Comment:
Do we have any models for which `id` **is** part of the q filter? That would seem odd to me.

Reply:
When testing it appears that `id` was dynamically getting inserted even when not declared.

Reply:
Ah right. SearchFilter always includes an `id: iexact` predicate. I'd forgotten that.